docs/design-documents/README.md

```diff
@@ -0,0 +1,44 @@
+
+## Design documents location and organization
+* It's suggested to write one design document per change, which could be a new feature, defect fixes, tools changes etc. Do not try to address multiple changes in the same document.
+* It's up to the judgement of developer to decide whether a change warrant a design document to go with it. For example, minor changes/small localized defect fixes may not need a design document.
+* The design document will follow the format represented in [Design Document Template](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/design_template.md)
+
+* For each design submission, create a folder under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}*. The area/component would be Drivers, Platform, Connectivity, Security, Peripherals etc. 
+  If your implementation is creating new APIs, you should try to match the folder name directly to the new API, otherwise 
+  use the title of the feature as the folder name. 
+  For example, if your implementation is adding new EMAC API, you may name the folder as **New_EMAC_API** or if you are creating a new Serial Driver for Mbed-OS you may name the folder as **Serial_driver_for_Mbed-os**.
+* Your design document itself should also be named with the same name of the as the folder, so that it serves
+  as your master design document, if you happen to use other .md files under same folder as annexures or reference.
+* The diagrams you are using may exist in the sub-folder *diagrams* under the new folder you have created.
+  See [Example Design Document](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/example_feature_design/example_feature_design.md) for example design document.
+
+### Using diagrams in Design documents
+Its highly encouraged to create diagrams to depict your software design. The [Design Document Template]((https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/design_template.md) provided here uses diagrams generated using draw.io. draw.io generates XMLs for each diagram and those corresponding XMLs are also provided under the diagrams folder, and you are free to use that as your reference for generating your own diagrams. And, when you are generating your own diagrams it would be better to provide the metadata/other files(e.g, if you are using draw.io provide XMLs from draw.io) required for future editing of those diagrams.
+
+## Design Process
+* Developer generates a design document located under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}* and generates a pull-request to Mbed-OS.
+  * NOTE: If you are part of an organization and would like to do design review within your organization before landing the pull-request, its ok to do so.
+  * The pull request may also include the initial code implementation. It's upto to the develper to decide if the initial implementation comes along with the design document or later.
```

Comment:
I would mention that it's advisable to start with a basic outline of an idea, so people can discuss it and then refine it.

Reply:
I feel like these should be two separate PRs - otherwise you'll have to update the code implementation in sync with each change to the spec.

For NFC we have a PR up for the spec + an internal PR with the API. We rebase the API branch regularly against the spec branch.